chore: bump actions/checkout from v5 to v6

somethingwithproof opened the pull request:
## Summary

This PR updates the `actions/checkout` action from v5 to v6 in all GitHub Actions workflows.

## Changes

- Update `actions/checkout@v5` → `actions/checkout@v6`

## Benefits

- Performance improvements
- Better support for newer GitHub features
- Keeps dependencies up to date
- Security improvements from the latest version

## Testing

The workflows will be automatically tested when this PR is opened. All existing functionality should continue to work as expected since v6 is backward compatible with v5.

## References

- [actions/checkout releases](https://github.com/actions/checkout/releases)

thomasvincent commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/thomasvincent/pr-checks/b764ec99-5f5d-4ba0-9897-1441f7b4ddc1?refs=fadc9fb568a9b62af256bdf48dabe49464a648e3&source=prChecksComment)

somethingwithproof committed:
chore: bump actions/checkout from v5 to v6

Update GitHub Actions checkout action to the latest major version.

Changes:
- Update actions/checkout@v5 to actions/checkout@v6

The v6 release includes performance improvements and better support
for newer GitHub features.
